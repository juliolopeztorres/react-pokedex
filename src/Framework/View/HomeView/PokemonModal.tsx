import React from 'react'
import { hot } from "react-hot-loader/root";

const PokemonModal: (props: {name: string, onPokemonArtClicked: () => void}) => React.ReactNode = ({name, onPokemonArtClicked}) => {

  return <div className="z-50" aria-labelledby="modal-title" role="dialog" aria-modal="true">
    <div className="fixed inset-0 bg-gray-500 dark:bg-gray-700 bg-opacity-75 dark:bg-opacity-90 transition-opacity"/>
    <div className="fixed inset-0 z-50 overflow-y-auto">
      <div className="flex min-h-full items-center justify-center text-center p-4 sm:p-0">
        <div className="rounded-lg bg-white">
          <div className="text-end mt-1">
            <span data-test-id='closeBtn' className='text-black text-xl font-mono hover:text-orange-600 mr-3' style={{cursor: 'pointer'}} onClick={() => onPokemonArtClicked()}>x</span>
          </div>
          <img className='px-4 pb-2' src={'https://img.pokemondb.net/artwork/' + name + '.jpg'} alt="Pokemon image"/>
          <p className="italic text-gray-400 text-xs text-center mb-2">.jpg art with white background</p>
        </div>
      </div>
    </div>
  </div>
}

export default hot(PokemonModal)
