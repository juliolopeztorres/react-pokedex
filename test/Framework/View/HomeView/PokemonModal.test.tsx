import React from 'react'
import { createRoot, Root } from "react-dom/client"
import { act } from "@testing-library/react";
import PokemonModal from "../../../../src/Framework/View/HomeView/PokemonModal";

let root: Root
let container: HTMLElement
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div")
  document.body.appendChild(container)

  root = createRoot(container)
})

afterEach(() => {
  // cleanup on exiting
  act(() => root.unmount())

  container.remove()
})

it('should render', () => {
  const onPokemonArtClicked = jest.fn()

  act(() => {
    root.render(<PokemonModal name={'bulbasaur'} onPokemonArtClicked={onPokemonArtClicked}/>)
  })

  expect(container).toMatchSnapshot()

  const closeBtnContainer = container.querySelector('span[data-test-id="closeBtn"]')
  expect(closeBtnContainer).toBeTruthy()

  expect(onPokemonArtClicked).not.toHaveBeenCalled()
  act(() => {
    closeBtnContainer!.dispatchEvent(new MouseEvent('click', {bubbles: true}))
  })

  expect(onPokemonArtClicked).toHaveBeenCalledTimes(1)
})
