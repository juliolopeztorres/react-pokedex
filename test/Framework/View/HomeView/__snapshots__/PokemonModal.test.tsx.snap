// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    aria-labelledby="modal-title"
    aria-modal="true"
    class="z-50"
    role="dialog"
  >
    <div
      class="fixed inset-0 bg-gray-500 dark:bg-gray-700 bg-opacity-75 dark:bg-opacity-90 transition-opacity"
    />
    <div
      class="fixed inset-0 z-50 overflow-y-auto"
    >
      <div
        class="flex min-h-full items-center justify-center text-center p-4 sm:p-0"
      >
        <div
          class="rounded-lg bg-white"
        >
          <div
            class="text-end mt-1"
          >
            <span
              class="text-black text-xl font-mono hover:text-orange-600 mr-3"
              data-test-id="closeBtn"
              style="cursor: pointer;"
            >
              x
            </span>
          </div>
          <img
            alt="Pokemon image"
            class="px-4 pb-2"
            src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
          />
          <p
            class="italic text-gray-400 text-xs text-center mb-2"
          >
            .jpg art with white background
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`
